Assignment/PingProcess.cs

```diff
@@ -21,47 +21,45 @@ public PingResult Run(string hostNameOrAddress)
         StringBuilder? stringBuilder = null;
         void updateStdOutput(string? line) =>
             (stringBuilder??=new StringBuilder()).AppendLine(line);
-        Process process = RunProcessInternal(StartInfo, updateStdOutput, default, default);
-        return new PingResult( process.ExitCode, stringBuilder?.ToString());
-    }
-
-    public Task<PingResult> RunTaskAsync(string hostNameOrAddress)
-    {
-        throw new NotImplementedException();
+        Process process = RunProcessInternal(StartInfo, updateStdOutput, default, default); 
+        string stdOutput = stringBuilder?.ToString() ?? string.Empty; 
+        if (string.IsNullOrWhiteSpace(stdOutput)) 
+        { 
+            stdOutput = $"Ping request could not find host {hostNameOrAddress}. Please check the name and try again."; 
+        } 
+        return new PingResult(process.ExitCode, stdOutput); 
+    } 
+    public Task<PingResult> RunTaskAsync(string hostNameOrAddress, CancellationToken cancellationToken = default) 
+    { 
+        return Task.Run(() => Run(hostNameOrAddress)); 
     }
 
     async public Task<PingResult> RunAsync(
         string hostNameOrAddress, CancellationToken cancellationToken = default)
     {
-        Task task = null!;
-        await task;
-        throw new NotImplementedException();
+        return await Task.Run(() =>
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            return Run(hostNameOrAddress);
+        }, cancellationToken);
     }
 
     async public Task<PingResult> RunAsync(params string[] hostNameOrAddresses)

```

Comment:
This modifier declaration order is inconsistent from the standard. It should be:
```suggestion
    public async Task<PingResult> RunAsync(params string[] hostNameOrAddresses)

```